master/buildbot/steps/download_secret_to_worker.py

```diff
@@ -0,0 +1,87 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+from __future__ import absolute_import
+from __future__ import print_function
+
+import os
+
+from twisted.internet import defer
+
+from buildbot.process.buildstep import FAILURE
+from buildbot.process.buildstep import SUCCESS
+from buildbot.steps.transfer import _TransferBuildStep
+from buildbot.steps.worker import CompositeStepMixin
+
+
+class WorkerFileSecret(object):
+
+    def __init__(self, path, secretvalue):
+        self.path = path
+        if not isinstance(self.path, str):
+            raise ValueError("Secret path %s is not a string" % path)
+        self.secretvalue = secretvalue
+        if not os.path.exists(os.path.dirname(self.path)):
+            raise ValueError("Path %s does not exist")
+
+
+class DownloadSecretsToWorker(_TransferBuildStep, CompositeStepMixin):
```

Comment:
I dont think it needs to inherit transferbuildstep. BuildStep should be enough + CompositeStepMixin of course